### ISSUE TITLE
CS9282 reported for readonly static property on generic type

### ISSUE THREAD

**Issue by drieseng** [dotnet/roslyn#82239]
**Version Used**: 
Visual Studio v18.2.1

**Steps to Reproduce**:

```c#
using System.Collections.Generic;
using System.Collections.ObjectModel;

public static class ReadOnlyDictionaryExtensions
{
    extension<TKey, TValue>(ReadOnlyDictionary<TKey, TValue>)
        where TKey : notnull
    {
        public static ReadOnlyDictionary<TKey, TValue> NewEmpty { get; } = new ReadOnlyDictionary<TKey, TValue>(new Dictionary<TKey, TValue>());
    }
}
```

**Diagnostic Id**:

ReadOnlyDictionaryExtensions.cs(9,56,9,61): error CS9282: This member is not allowed in an extension block

**Expected Behavior**:
No error is reported.

**Actual Behavior**:
CS9282 is reported.

**Comment by RikkiGibson** [comment]
It looks like it is probably complaining about the auto-property here. @jcouv I imagine inability to use static fields in extensions is a known limitation?

This seems like a slightly painful case because if you want to manually declare the backing field outside the extension, you need to declare a new generic type to hold the field also.

**Comment by jcouv** [comment]
That's correct. Fields (static or otherwise) are disallowed in extension blocks
